### PR TITLE
DT-1773 Move all building pages to new validator with a non-async ver…

### PR DESCRIPTION
**Pull request by andymarke**
…sion

### CODE REVIEW

File: server/validation/validation.ts [hunk 1]
```diff
@@ -13,22 +26,9 @@ export default function validateAsync<T>(
...
-      }
-      return false
-    },
-    'Enter a real postcode, like AA11AA'
-  )
-
   Validator.registerAsync(
     'unique-subcode',
-    async (value, requirement, request, passes) => {
+    async (value, allowedBuildingId, request, passes) => {
```
Comment (andymarke) [inline server/validation/validation.ts:31]: 2nd parameter is essentially a config type parameter

File: server/routes/courtRegister/amendCourtBuildingValidator.ts [hunk 2]
```diff
@@ -17,50 +10,31 @@ export default async function validate(
...
+      subCode: ['between:0,6', `unique-subcode:${form.id}`],
+    },
+    {
+      'required.buildingname': 'Enter the building name',
+      'required.addressline1': 'Enter the first line of the address',
+      'required.addresstown': 'Enter the town or city',
+      'required.addresscounty': 'Enter the county',
+      'required.addresspostcode': 'Enter the postcode, like AA11AA',
+      'required.addresscountry': 'Select the country',
+      'between.subCode': 'Enter a unique building code not greater than 6 characters',
```
Comment (andrewrlee) [inline server/routes/courtRegister/amendCourtBuildingValidator.ts:31]: so the custom validator doesn't require a message as it's hardcoded in there 👍

File: server/routes/courtRegister/amendCourtBuildingValidator.ts [hunk 3]
```diff
@@ -17,50 +10,31 @@ export default async function validate(
...
+  const errors = await validateAsync(
+    form,
+    {
+      buildingname: 'required',
+      addressline1: 'required',
+      addresstown: 'required',
+      addresscounty: 'required',
+      addresspostcode: ['required', 'postcode'],
+      addresscountry: 'required',
+      subCode: ['between:0,6', `unique-subcode:${form.id}`],
```
Comment (andrewrlee) [inline server/routes/courtRegister/amendCourtBuildingValidator.ts:22]: does the framework eval this? or is it a mini embedded dsl?

**Review by andrewrlee — APPROVED** [review]
LGTM

File: server/routes/courtRegister/amendCourtBuildingValidator.ts (hunk 2 above)
Comment (andymarke) [reply server/routes/courtRegister/amendCourtBuildingValidator.ts:31]: correct. My custom validator has the a suitable error message out of the box.

File: server/routes/courtRegister/amendCourtBuildingValidator.ts (hunk 3 above)
Comment (andymarke) [reply server/routes/courtRegister/amendCourtBuildingValidator.ts:22]: For their inbuilt ones that parse the stuff after the colon. For custom you will receive and extra parameter with the righthand side of the colon. It is up to you to decide what it means.

**Pull request merged**